### PR TITLE
java_keystore/fix 1667 improve temp files storage

### LINKED ISSUE #1667: java_keystore: improve temp file storage
##### SUMMARY
The module writes everything (certificate, private key, intermediate PKCS12 file) to fixed filenames in /tmp/, and actually keeps the PKCS12 temp file it creates there.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
java_keystore

##### ANSIBLE VERSION
```paste below
all
```

### PR DESCRIPTION
**Pull request by quidame**
##### SUMMARY

This PR covers temporary files handling:
- files containing certificate, private key and pkcs12 are created with `tempfile.mkstemp()`
- temporary pkcs12 is removed as well as other materials

Fixes #1667 

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

java_keystore

### CODE REVIEW

**Comment by quidame** [comment]
hmm... unit tests fail due to fixed filenames (`/tmp/foo.crt`, `/tmp/foo.key`, `/tmp/keystore.p12`) in `test_java_keystore.py`, where the module uses random filenames (from `tempfile.mkstemp()`; that is the purpose of this PR) rather than the fixed filenames. And I really don't know how to deal with that.

Help is welcome :)

**Comment by russoz** [comment]
Hi @quidame the test output shows:
```
00:24 E           AssertionError: Expected call: mock(cmd=['openssl', 'x509', '-noout', '-in', '/tmp/foo.crt', '-fingerprint', '-sha256'], err='Oops', msg='', rc=1)
00:24 E           Actual call: mock(cmd=['openssl', 'x509', '-noout', '-in', <MagicMock name='create_file()' id='93862131003024'>, '-fingerprint', '-sha256'], err='Oops', msg='', rc=1)
```
we can see that `create_file()` is being patched properly, but what you are missing is setting up a `return_value` for that patch. And, personally, instead of creating the patch object in `setUp()`, I would rather go local in the test function, as in `tests/unit/plugins/modules/system/test_java_keystore.py` line 201 (your branch).

File: changelogs/fragments/2163-java_keystore_1667_improve_temp_files_storage.yml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+bugfixes:
+  - java_keystore - use tempfile lib to create temporary files with randomized
+    names, and remove the temporary pkcs12 keystore as well as other materials
+    (https://github.com/ansible-collections/community.general/issues/1667).
```
Comment (felixfontein) [inline changelogs/fragments/2163-java_keystore_1667_improve_temp_files_storage.yml:5]:
```suggestion
  - "java_keystore - use tempfile lib to create temporary files with randomized
    names, and remove the temporary PKCS#12 keystore as well as other materials
    (https://github.com/ansible-collections/community.general/issues/1667)."
```

**Comment by felixfontein** [comment]
I think with what @russoz wrote you should be able to fix the tests. Once they pass, I think this is ready for merging.

**Comment by quidame** [comment]
Fixed. Remaining errors are all about python 2.6 packages not found.

**Comment by felixfontein** [comment]
@quidame these are caused by PyPi's SNI brownouts... I've restarted them (outside the brownout time) and they passed.

**Review by felixfontein — APPROVED** [review]

**Comment by felixfontein** [comment]
@russoz any comments from your side?

**Comment by russoz** [comment]
nop, LGTM

**Comment by quidame** [comment]
@felixfontein @russoz thanks for reviewing and pointing me the right way !

**Pull request merged**